content/source/docs/enterprise/system-overview/architecture.html.md

```diff
@@ -5,24 +5,201 @@ page_title: "Architecture Summary - System Overview - Terraform Enterprise"
+~> **Note:** Several services described below include the legacy string `ptfe` as part of their names. See [here](https://www.terraform.io/docs/enterprise/index.html#note-about-product-names) for more details.
 
-These are the services used to run Terraform Enterprise. Each service contains a description of what actions it performs, a policy for restarts, impact of failing or degraded performance, and the service's dependencies.
+## Application Layer
+
+The application layer is divided conceptually into two areas, each with specific responsibilities:
+
+- Web Services - Answer requests forwarded by NginX and enqueue jobs for later
+- Background Services - Asynchronously process jobs enqueued by web services
+
+Most requests in Terraform Enterprise follow a similar lifecycle as shown in the diagram below.
+
+![tfe-int-app-layer](assets/tfe-int-app-layer.png)
+
+A web service receives a request and enqueues a background job to perform the task via the Coordination Layer (see below).  The web service then responds immediately to the original request instead of waiting for the job to be finished.
+
+A background service later claims the job from the Coordination Layer and performs the resource- or time-intensive task in the background without blocking the web service.
+
+Finally, the original web service is notified when then task is complete via an internal webhook.
+
+### Web Services
+
+These services answer requests from the UI and API as well as incoming webhooks from a connected VCS.
+
+### Background Services
+
+This is the largest part of the Application Layer and includes multi-purpose Sidekiq workers, Go services that provide key functionality of Terraform Enterprise, and a dynamic fleet of isolated execution environments that perform Terraform runs.
+
+## Coordination Layer
+
+The Coordination Layer is responsible for queuing asynchronous jobs and distributing them to background services running in the Application Layer. Standalone deployments will start a local Redis container for this purpose and Active/Active deployments will use an external Redis service.
+
+![tfe-int-coordination-layer](assets/tfe-int-coordination-layer.png)
+
+~> **Note:** While these components are critical to the overall operation of Terraform Enterprise, operators rarely (if ever) need to look at them directly.
+
+## Storage Layer
+
+All application data is stored in PostgreSQL and object storage. If you have selected Mounted Disk mode, local services will be started for you, otherwise, the appropriate services will be configured to access a external PostgreSQL, S3-compatible blob storage, and, in active-active, Redis.
+
+~> **Note:** The configuration information provided and/or generated during installation (e.g. database credentials, hostname, etc.) is stored outside of the application.
+
+## Key Component Interactions
+
+This section details how the key containers work together to accomplish common tasks.
+
+### Initial Startup - External Services Mode
```

Comment:
In all of these sections, I'd move the steps to be above the diagram. I think that will be more cohesive for users (to set them up for what they're observing in the diagram, and then we also won't split sentences that say "the following steps" from the following steps.